fix: temperature and date display formats

mairas opened the pull request:
Celsius and Fahrenheit units were missing the degree mark (°). This has been fixed.

The selection of allowed date formats was IMO arbitrary and limited. It has been fixed to be more internationally representative (although still arbitrary - I tried to cover all common cases I found, though). This was slightly opinionated - I also dropped the two-digit-year variants because I don't think those are the official format in any locale.

cmotelet commented:
About the date, the new formats are nice.
For the `DD/MM/YY` format, maybe it's because before Y2K I worked on a lot of text consoles and I saw this format very often, for me it's very familiar.
I think it would be good to keep it even if it's only me who uses it.

cmotelet commented:
About the temperature unit, this is a breaking change.
![image](https://user-images.githubusercontent.com/31212553/229342577-689d870a-7a12-4c37-97cb-7fae5e791ee6.png)
The conversion unit is stored in the widget configuration and also in the preferred units page.
If 'C' and 'F' don't exist anymore it breaks the compatibility after the update.
I think you should keep 'C' and 'F' and add the new units.
Can you also add in `basewidget.js` after line 212 `this.options.convertTo = ''`

tkurki commented:
Would adding the degree in basewidget.getFinalUnit work?

cmotelet commented:
@tkurki , yes, I had thought of proposing it at this place but it obliges to add an 'if' to process it.
 @mairas , the display, which I grant you is wrong, has been like this for almost 7 years and I don't think anyone has been scalded with this 'C' or this 'F'
 That's why I proposed a smooth transition by keeping the 2 display formats for some time, maybe years.
 This does not require any specific code and will not disturb users who have already set up their widgets.

mairas commented:
Sorry, didn't mean to be abrasive - I just think display units and formats have a major impact on the quality impression of software, especially when they play such a major role as in SK. I personally have avoided using the instrumentpanel precisely due to such issues and finally wanted to fix them.

In my opinion, I'd rather decisively fix such issues than make conscious decisions to keep them alive. I've now added a simple `unitCorrections` object that maps the incorrect units to the correct ones and updates the `options.convertTo` field in `getFinalUnit`. I hope this addresses your concerns!

cmotelet commented:
perfect it's a good compromise
